migration/src/m20240124_181900_add_slot_updated_column_per_update_type.rs

```diff
@@ -0,0 +1,159 @@
+use sea_orm_migration::{
+    prelude::*,
+    sea_orm::{ConnectionTrait, DatabaseBackend, Statement},
+};
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let connection = manager.get_connection();
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+            ALTER TABLE asset
+            ADD COLUMN slot_updated_metadata_account int8;
+            "
+                .to_string(),
+            ))
+            .await?;
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+            ALTER TABLE asset
+            ADD COLUMN slot_updated_token_account int8;
+            "
+                .to_string(),
+            ))
+            .await?;
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+            ALTER TABLE asset
+            ADD COLUMN slot_updated_mint_account int8;
+            "
+                .to_string(),
+            ))
+            .await?;
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+            ALTER TABLE asset
+            ADD COLUMN slot_updated_cnft_transaction int8;
+            "
+                .to_string(),
+            ))
+            .await?;
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+                CREATE OR REPLACE FUNCTION update_slot_updated()
+                RETURNS TRIGGER AS $$
+                BEGIN
+                   NEW.slot_updated = GREATEST(NEW.slot_updated_token_account, NEW.slot_updated_mint_account, NEW.slot_updated_metadata_account, NEW.slot_updated_cnft_transaction);
+                   RETURN NEW;
+                END;
+                $$ language 'plpgsql';
+                "
+                    .to_string(),
+            ))
+            .await?;
+
+        connection
+            .execute(Statement::from_string(
+                DatabaseBackend::Postgres,
+                "
+                CREATE TRIGGER update_slot_updated_trigger
+                BEFORE UPDATE ON asset
+                FOR EACH ROW
+                EXECUTE PROCEDURE update_slot_updated();
+                "
+                .to_string(),
+            ))
+            .await?;
```

Comment:
I tested it out and it didn't seem to substantially affect indexing speed. There are a ton of updates happening during metadata account ingestion already.  

Slot updated will get updated in multiple different places and will get updated in even more places as the code evolves. So I found it a bit repetitive and error prone to add this logic everywhere instead of using a trigger. I do feel your points are valid, but it does seem like a tradeoff. If you feel strongly about this still, then I can remove the trigger.